s2fft/precompute_transforms/construct.py

```diff
@@ -53,27 +60,102 @@ def spin_spherical_kernel(
         )
+    if mode.lower() == "price_mcewen" and abs(spin) >= 5:
+        mode = "risbo"
+        warn(
+            "The Price-McEwen can become unstable for spins >= 5. "
+            + "Defaulting to slower but more stable Risbo recursions."
+        )
+
+    if mode.lower() == "auto":
+        # This mode automatically determines which recursion is best suited for the
+        # current parameter configuration.
+        mode = "risbo" if abs(spin) >= 5 else "price_mcewen"
+
+    dl = []
     m_start_ind = L - 1 if reality else 0
     m_dim = L if reality else 2 * L - 1
 
+    # Determine theta locations for forward vs inverse transform.
     if forward and sampling.lower() in ["mw", "mwss"]:
         sampling = "mwss"
         thetas = samples.thetas(2 * L, "mwss")
     else:
         thetas = samples.thetas(L, sampling, nside)
 
-    dl = np.zeros((len(thetas), L, m_dim), dtype=np.float64)
-    for t, theta in enumerate(thetas):
-        for el in range(abs(spin), L):
-            dl[t, el] = recursions.turok.compute_slice(theta, el, L, -spin, reality)[
-                m_start_ind:
-            ]
-            dl[t, el] *= np.sqrt((2 * el + 1) / (4 * np.pi))
+    # Calculate Wigner d-function elements through the Price-McEwen recursion.
+    # - The complexity of this approach is O(L^3).
+    # - This approach becomes inaccurate for abs(spins) >= 5.
+    if mode.lower() == "price_mcewen":
+        dl = np.zeros((len(thetas), L, m_dim), dtype=np.float64)
+        for t, theta in enumerate(thetas):
+            for el in range(abs(spin), L):
+                dl[t, el] = recursions.turok.compute_slice(
+                    theta, el, L, -spin, reality
+                )[m_start_ind:]
+                dl[t, el] *= np.sqrt((2 * el + 1) / (4 * np.pi))
+
+    # Calculate Wigner d-function elements through the Risbo recursion.
+    elif mode.lower() == "risbo":
+        dl = np.zeros((len(thetas), L, m_dim), dtype=np.float64)
+
+        # GL and HP sampling ARE NOT uniform in theta therefore CANNOT be calculated
+        # using the Fourier decomposition of Wigner d-functions. Instead they must be
+        # manually bootstrapped from the recursion.
+        # - The complexity of this approach is O(L^4).
+        # - This approach is stable for arbitrary abs(spins) <= L.
+        if sampling.lower() in ["healpix", "gl"]:
+            delta = np.zeros(
+                (len(thetas), 2 * L - 1, 2 * L - 1), dtype=np.float64
+            )
+            for el in range(L):
+                delta = recursions.risbo.compute_full_vect(delta, thetas, L, el)
+                dl[:, el] = delta[:, m_start_ind:, L - 1 - spin]
+
+        # MW, MWSS, and DH sampling ARE uniform in theta therefore CAN be calculated
+        # using the Fourier decomposition of Wigner d-functions.
+        # - The complexity of this approach is O(L^3LogL).
+        # - This approach is stable for arbitrary abs(spins) <= L.
+        if sampling.lower() in ["mw", "mwss", "dh"]:
```

Comment:
This should be `elif`.  Also we should have a final `else` to catch any unsupported sampling schemes.

Reply:
This error is caught earlier in lower level functions to compute theta values. No need to duplicate higher up in the code I think?